malformed/test_malformed_headers.py

```diff
@@ -0,0 +1,487 @@
+from framework import tester
+from helpers import tf_cfg, deproxy
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2018 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class MalformedRequestsTest(tester.TempestaTest):
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' :
+"""HTTP/1.1 200 OK
+Content-Length: 0
+Connection: close
+
+"""
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+cache 0;
+listen 80;
+
+srv_group default {
+    server ${general_ip}:8000;
+}
+
+vhost default {
+    proxy_pass default;
+}
+""",
+    }
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+    ]
+
+    def common_check(self, request):
+        deproxy = self.get_server('deproxy')
+        deproxy.start()
+        self.start_tempesta()
+        self.assertTrue(deproxy.wait_for_connections(timeout=1))
+        deproxy = self.get_client('deproxy')
+        deproxy.start()
+        deproxy.make_request(request)
+        resp = deproxy.wait_for_response(timeout=5)
+        self.assertTrue(resp, "Response not received")
+        status = deproxy.last_response.status
+        self.assertEqual(int(status), 400, "Wrong status: %s" % status)
+
+    def test_accept(self):
+        # https://tools.ietf.org/html/rfc7231#section-5.3.2
+        request = 'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  'Accept: invalid\r\n' \
```

Comment:
The test incomplete. The RFC says that `Accept` header is a list of `media-range` and parameters. There are a lot of ways how no make the header invalid: e.g. `/` or `,` can be missed. And even more, most programming mistakes happen on borders, most mistakes in our parser happens when a term border happens at skb border. Unit tests for http parser contains very efficient fuzzer to cover the case, implementing the same thing in functional tests is too complicated and unreliable. As for me tempesta-tech/tempesta#822 cant be solved at functional tests level, unit tests are also required. I've raised this question in chat when this PR was introduced, but I don't see that it was taken into account. 

Here is an example how the `Cookie:` header parsing is checked in unit tests: https://github.com/tempesta-tech/tempesta/blob/6adac2cf84e923d671644930e05cd7879b3f5cde/tempesta_fw/t/unit/test_http_parser.c#L1251 It shows the idea. 

Same apply to other headers checks.

Reply:
Yes, there are a lot of ways for make invalid header. I think, thar full test for each header format should be really done in the unit tests. But the purpose of this functional tests is not a full-covering parser check, but checking that parser and whole tempesta make necessary checks

full-covering test for parsers is a different task